src/scripts/main.js

```diff
@@ -1,3 +1,154 @@
 'use strict';
 
-// write code here
+const tableBody = document.querySelector('tbody');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const headings = document.querySelectorAll('thead th');
+
+function convertToNumber(string) {
+  return parseInt(string.slice(1).replaceAll(',', ''));
+}
+
+let columnIndex;
+let isAscending = true;
+
+headings.forEach(heading => {
+  heading.addEventListener('click', e => {
+    const index = e.target.cellIndex;
+
+    if (columnIndex === index) {
+      isAscending = !isAscending;
+    } else {
+      columnIndex = index;
+      isAscending = true;
+    }
+
+    rowsArray.sort((a, b) => {
+      const aText = a.cells[index].textContent;
+      const bText = b.cells[index].textContent;
+      const callWithA = convertToNumber(aText);
+      const callWithB = convertToNumber(bText);
+
+      if (isAscending) {
+        if (+aText) {
+          return +aText - (+bText);
+        } else if (callWithA) {
+          return callWithA
+          - callWithB;
+        } else if (typeof aText === 'string'
+        && !callWithA) {
+          return aText.localeCompare(bText);
+        }
+      } else {
+        if (+aText) {
+          return +bText - (+aText);
+        } else if (callWithA) {
+          return callWithB
+          - callWithA;
+        } else if (typeof aText === 'string'
+        && !callWithA) {
+          return bText.localeCompare(aText);
+        }
+      }
+    });
+    tableBody.append(...rowsArray);
+  });
+});
+
+rowsArray.forEach(row => {
+  row.addEventListener('click', () => {
+    rowsArray.forEach(tr => {
+      tr.classList.remove('active');
+    });
+    row.classList.add('active');
+  });
+});
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const options = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+function createInputField(text, tag, type) {
+  const element = document.createElement(`${tag}`);
+
+  if (`${tag}` === 'button' || `${tag}` === 'input') {
+    element.type = `${type}`;
+  }
+
+  if (`${tag}` === 'button') {
+    element.textContent = `${text}`;
+
+    return element;
+  }
+
+  const label = document.createElement(`label`);
+
+  label.textContent = `${text}`;
+
+  const labelText = label.textContent;
+  const textModified = labelText
+    .slice(0, labelText.indexOf(':')).toLowerCase();
+
+  element.dataset.qa = textModified;
+  element.name = textModified;
+  // element.required = true;
+
+  if (`${tag}` === 'select') {
```

Comment:
Avoid using a `for...of` loop to create a new array or append elements to an array. Instead, use `map` or `forEach`.